File: Semester_3/DSAL/lab3/q1.c
#include<stdio.h>
#include<stdlib.h>

typedef struct
{
	float real;
	float img;
}complex;

complex sum(complex a, complex b)
{
	complex s;
	s.real = a.real + b.real;
	s.img = a.img + b.img;
	return s;
}

complex diff(complex a, complex b)
{
	complex d;
	d.real = a.real - b.real;
	d.img = a.img - b.img;
	return d;
}

complex product(complex a, complex b)
{
	complex m1;
	m1.real = (a.real * b.real) - (a.img * b.img);
	m1.img = (a.img * b.real) + (a.real * b.img);
	return m1;
}

int main()
{
	complex a, b;

	printf("Enter the real and imaginary parts of complex number 1: \n");
	scanf("%f%f", &a.real, &a.img);
	printf("Enter the real and imaginary parts of complex number 2: \n");
	scanf("%f%f", &b.real, &b.img);

	printf("Sum of the two complex numbers: %.2f + %.2fi \n", sum(a,b).real, sum(a,b).img);
	printf("Difference of the two complex numbers: %.2f + %.2fi \n", diff(a,b).real, diff(a,b).img);
	printf("Product of the two complex numbers: %.2f + %.2fi \n", product(a, b).real, product(a, b).img );

	return 0;
}



  

File: Semester_3/OOP/week3/q1.java

class Complex
{
    int r;
    int img;

    void setvalues(int real , int imgaginary )
    {
        r = real;
        img = imgaginary;
    }
    Complex sum(Complex n1, Complex n2)
    {
        Complex temp = new Complex();
        temp.r = n1.r + n2.r;
        temp.img = n1.img + n2.img;
        return temp;
    }
    Complex diff(Complex n1, Complex n2)
    {
        Complex temp = new Complex();
        temp.r = n1.r -  n2.r;
        temp.img = n1.img - n2.img;
        return temp;
    }
    void print()
    {
        System.out.println("Complex number: " + r + " + " + img + "i");
    }
}

class q1
{
    public static void main(String[] args)
    {
        Complex n1 = new Complex();
        n1.setvalues(5, 4);
        n1.print();
        Complex n2 = new Complex();
        n2.setvalues(7, 3);
        n2.print();
        Complex n3 = new Complex();
        n3 = n3.sum(n1, n2);

        System.out.println("Sum of ");
        n3.print();

        n3 = n3.diff(n1, n2);

        System.out.println("Difference of ");
        n3.print();
    }
}
